meta_test.go

```diff
@@ -964,6 +964,102 @@ extra-bindings:
 	})
 }
 
+func (s *MetaSuite) TestDevice(c *gc.C) {
+	meta, err := charm.ReadMeta(strings.NewReader(`
+name: a
+summary: b
+description: c
+device:
+    nvidia.com/gpu:
```

Comment:
This is the type
The name is "bitcoin-miner" or something meaningful to the charm